### PR TITLE
Fix sulu.loadUserSettings() when key does not exist

### PR DESCRIPTION
**Pull request by rupkoe**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | yes |
| Deprecations? | no |
| Fixed tickets | - |
| Related issues/PRs | - |
| License | MIT |
| Documentation PR | - |
#### What's in this PR?

Removes faulty sulu.loadUserSetting() method.
#### Why?

As (partially) discussed with @danrot in https://sulu-io.slack.com/archives/general/p1476195014001682 

Method has two problems when key does not yet exist in the DB (and also is not in the dynamic settings in the sandbox):
1. Wrong call to log method.
2. Callback is not called when key does not exist in neither sandbox nor DB. This gives a hard time when trying to use method in promises (like with the loadComponentData hook). 

As method is considered deprecated (in favor of sulu.getUserSetting()) it is removed completely.

### CODE REVIEW

File: src/Sulu/Bundle/AdminBundle/Resources/public/js/aura_extensions/sulu-extension.js [hunk 1]
```diff
@@ -94,7 +94,8 @@
                             callback(data);
                         }.bind(this))
                         .fail(function(data) {
-                            app.sandbox.logger.log('data could not be loaded:', data);
+                            app.logger.log('data could not be loaded:', data);
```
Comment (danrot) [inline src/Sulu/Bundle/AdminBundle/Resources/public/js/aura_extensions/sulu-extension.js]: I have to check this in more detail... When i look at it this way the statement a few lines above should also not be working (`app.*sandbox*.sulu.userSettings`).

File: src/Sulu/Bundle/AdminBundle/Resources/public/js/aura_extensions/sulu-extension.js [hunk 2]
```diff
@@ -94,7 +94,8 @@
...
+                            callback(null);
```
Comment (danrot) [inline src/Sulu/Bundle/AdminBundle/Resources/public/js/aura_extensions/sulu-extension.js]: Have to make sure that this is not a BC break. If the previous behavior was really that the application was breaking it should not be such a big deal.

**Comment by danrot** [comment]
A line in the changelog is missing. Please have a look at the others and add one.

File: src/Sulu/Bundle/AdminBundle/Resources/public/js/aura_extensions/sulu-extension.js (hunk 2 above)
Comment (rupkoe) [inline src/Sulu/Bundle/AdminBundle/Resources/public/js/aura_extensions/sulu-extension.js]: I wondered if it's a BC or not, too. But then I see it as a bug in the first place: uncaught type exception, and no callback called ever, although this is a none-error situation. Probably in everyday usage it does not occur very often: you'd need to load settings which haven't been set ever before to have this actually kick in. Or maybe a broken backend would do it, too.

**Comment by rupkoe** [comment]
Hi @danrot - any updates on that? Would be good to know if we might have a chance to wait for the fix, or if we have to find a way to code around this.

**Comment by wachterjohannes** [comment]
@rupkoe in my opinion this fix should be included in 1.4 (except @danrot knows bigger disadvantages). The problem is that danrot is not callable in the next days and i have no good feelings merging that without talking to him. So we should wait for him. But you could use this your fork in the mean time or is that not possible for you?

**Comment by rupkoe** [comment]
@wachterjohannes thanks for the update. Using the fork might be an option then, knowing that the fix will arrive in the official release eventually.

**Comment by danrot** [comment]
From what I see this method is not used at all... I guess it is deprecated... We are actually using `this.sandbox.sulu.getUserSetting` and `this.sandbox.sulu.saveUserSetting` in our code base. It seems that this is some deprecated code. 

Would the two mentioned methods work for you? If you could use the other ones I would suggest doing a PR which removes the method you have just touched. I know it a is a BC break, but it would be much cleaner.

**Comment by rupkoe** [comment]
Checked the suggested method, I can adjust the code to work with it instead. 

I think the faulty method is quite misleading, so I'm happy to update the PR to remove it entirely.

**Comment by rupkoe** [comment]
I updated this PR according to our discussion, that's okay for you?

**Comment by danrot** [comment]
That's ok for me, if @chirimoya or @wachterjohannes do not have any doubts we can merge that.

Could you also resolve the conflicts?

File: CHANGELOG.md [hunk 3]
```diff
@@ -15,6 +17,9 @@ CHANGELOG for Sulu
     * BUGFIX      #2985 [ContentBundle]       Fixed link-provider overlay-spacing
     * BUGFIX      #2983 [AdminBundle]         Removed default placeholder for datepicker
     * FEATURE     #2979 [MediaBundle]         Hide internal formats in UI
+=======
+    * BUGFIX      #2977 [AdminBundle]         Removed deprecated method sulu.loadUserSettings()
+>>>>>>> Removed faulty method sulu.loadUserSetting entirely
```
Comment (danrot) [inline CHANGELOG.md]: You are having an unresolved conflict here :-)

Comment (rupkoe) [inline CHANGELOG.md]: Ups seems I rushed things a bit ... ;)

**Comment by chirimoya** [comment]
@turbo-ele ... please verify if `loadUserSetting` is used in your setup.

**Comment by turbo-ele** [comment]
Doesn't seem to be used anywhere. So it's okay for me ;)

**Comment by danrot** [comment]
@rupkoe Thanks for you contribution :-)

**Pull request merged**